src/server/kiwix-serve.cpp

```diff
@@ -41,50 +41,47 @@
-            << "\t-L, --ipConnectionLimit\tMax number of (concurrent) connections per IP (default: infinite, recommended: >= 6)" << std::endl
-            << "\t-k, --skipInvalid\tStartup even when ZIM files are invalid (those will be skipped)" << std::endl
-            << std::endl
-
-            << "Documentation:" << std::endl
-            << "\tSource code\t\thttps://github.com/kiwix/kiwix-tools" << std::endl
-            << "\tMore info\t\thttps://wiki.kiwix.org/wiki/Kiwix-serve" << std::endl
-            << std::endl;
-}
+#define LITERAL_AS_STR(A) #A
+#define AS_STR(A) LITERAL_AS_STR(A)
+
+
+static const char USAGE[] =
+R"(Deliver ZIM file(s) articles via HTTP
+
+Usage:
+ kiwix-serve [options] ZIMPATH ...
+ kiwix-serve [options] (-l | --library) LIBRARYPATH
+ kiwix-serve -h | --help
+ kiwix-serve -V | --version
+
+Mandatory arguments:
+  LIBRARYPATH  XML library file path listing ZIM file to serve. To be used only with the --library argument."
+  ZIMPATH      ZIM file path(s)
+
+Options:
+ -h --help                               Print this help
+ -a=<pid> --attachToProcess=<pid>        Exit if given process id is not running anymore [default: 0]
+ -d --daemon                             Detach the HTTP server daemon from the main process
+ -i=<address> --address=<address>        Listen only on the specified IP address. Specify 'ipv4', 'ipv6' or 'all' to listen on all IPv4, IPv6 or both types of addresses, respectively [default: all]
+ -M --monitorLibrary                     Monitor the XML library file and reload it automatically
+ -m --nolibrarybutton                    Don't print the builtin home button in the builtin top bar overlay
+ -n --nosearchbar                        Don't print the builtin bar overlay on the top of each served page
+ -b --blockexternal                      Prevent users from directly accessing external links
+ -p=<port> --port=<port>                 Port on which to listen to HTTP requests [default: 80]
+ -r=<root> --urlRootLocation=<root>      URL prefix on which the content should be made available [default: /]
+ -s=<limit> --searchLimit=<limit>        Maximun number of zim in a fulltext multizim search [default: 0]
+ -t=<threads> --threads=<threads>        Number of threads to run in parallel [default: )" AS_STR(DEFAULT_THREADS) R"(]
+ -v --verbose                            Print debug log to STDOUT
+ -V --version                            Print software version
+ -z --nodatealiases                      Create URL aliases for each content by removing the date
+ -c=<path> --customIndex=<path>          Add path to custom index.html for welcome page
+ -L=<limit> --ipConnectionLimit=<limit>  Max number of (concurrent) connections per IP [default: 0] (recommended: >= 6)
+ -k --skipInvalid                        Startup even when ZIM files are invalid (those will be skipped)
+
+Documentation:
+  Source code   https://github.com/kiwix/kiwix-tools
+  More info     https://wiki.kiwix.org/wiki/Kiwix-serve
```

Comment:
It makes sense to add the https://kiwix-tools.readthedocs.io/en/latest/kiwix-serve.html link